Detectors/MUON/MCH/Simulation/src/Digitizer.cxx

```diff
@@ -60,62 +60,76 @@ void Digitizer::init()
 
 void Digitizer::process(const std::vector<Hit> hits, std::vector<Digit>& digits)
 {
+  digits.clear();
+  mDigits.clear();
+  mMCTruthContainer.clear();
+  
   //array of MCH hits for a given simulated event
+  //to be added: exception handling
   for (auto& hit : hits) {
-    processHit(hit, mEventTime);
+    int labelIndex = mMCTruthContainer.getIndexedSize();//index for this hit
+    int detID = hit.GetDetectorID();
+    if(detID<299){
+      processHit(hit, detID, mMuonresponse_stat1, mEventTime, labelIndex);
+    } else {
+      processHit(hit, detID, mMuonresponse_stat2, mEventTime, labelIndex);
+    }
+ 
+    MCCompLabel label(hit.GetTrackID(), mEventID, mSrcID);
+    //Todo:: check/update DPL workflow how EventID and srcID set
+    mMCTruthContainer.addElementRandomAccess(labelIndex,label);//ToDo: check if needed like this
+
+    auto labels = mMCTruthContainer.getLabels(labelIndex);
+    std::sort(labels.begin(), labels.end());
   } // end loop over hits
-  //TODO: merge (new member function, add charge) of digits that are on same pad:
-  //things to think about in terms of time costly
+  //TODO: check handling of multiple hits per one pad
 
-  digits.clear();
   fillOutputContainer(digits);
 }
 
 //______________________________________________________________________
-int Digitizer::processHit(const Hit& hit, double event_time)
+int Digitizer::processHit(const Hit& hit, int detID, Response response, double event_time, int labelIndex)
 {
-
-  //hit position(cm)
+  //Question: what happens if several tracks producing signal on same pad up to now? -> should be additive? is this the case?
+  //assign the labelIndex to every digit that corresponds to
+  //this hit
+  //hit position(cm),hit has global coordinates
   Point3D<float> pos(hit.GetX(), hit.GetY(), hit.GetZ());
-  //hit has global coordinates
+  
   //convert energy to charge, float enough?
-  float charge = mMuonresponse.etocharge(hit.GetEnergyLoss());
-  //std::cout << "charge " << charge << std::endl;
-
+  float charge = response.etocharge(hit.GetEnergyLoss());
   //time information
   float time = hit.GetTime(); //to be used for pile-up
-  int detID = hit.GetDetectorID();
+
   //get index for this detID
   int indexID = mdetID.at(detID);
+
   //# digits for hit
   int ndigits = 0;
 
   //transformation from global to local
   auto t = o2::mch::getTransformation(detID, *gGeoManager);
   Point3D<float> lpos;
   t.MasterToLocal(pos, lpos);
-
-  float anodpos = mMuonresponse.getAnod(lpos.X(), detID);
-
+  float anodpos = response.getAnod(lpos.X());
   //TODO/Questions:
   //- possibility to do random seeding in controlled way?
   //distance of impact point and anode, needed for charge sharing
   float anoddis = TMath::Abs(pos.X() - anodpos);
```

Comment:
not used

Reply:
yes, put it should be needed in the proper implementation of the charge sharing (I don't agree with the current thing that is paste and copy from Aliroot). I can remove it or comment it if you prefer.